test/e2e/common_suite_test.go

```diff
@@ -237,6 +255,40 @@ func newTestCase(t *testing.T, testName string, assert CloudAssert, assessMessag
 	return testCase
 }
 
+func comparePodLogString(ctx context.Context, client klient.Client, customPod v1.Pod, expectedPodlogString string) (string, error) {
+	podLogString := ""
+	var podlist v1.PodList
+	clientset, err := kubernetes.NewForConfig(client.RESTConfig())
+	if err != nil {
+		return podLogString, errors.New(err.Error())
+	}
+	if err := client.Resources(customPod.Namespace).List(ctx, &podlist); err != nil {
+		return podLogString, errors.New(err.Error())
+	}
+	for _, pod := range podlist.Items {
+		if pod.ObjectMeta.Name == customPod.Name {
+			func() {
+				req := clientset.CoreV1().Pods(customPod.Namespace).GetLogs(pod.ObjectMeta.Name, &v1.PodLogOptions{})
+				podLogs, err := req.Stream(ctx)
+				if err != nil {
+					return
+				}
+				defer podLogs.Close()
+				buf := new(bytes.Buffer)
+				_, err = io.Copy(buf, podLogs)
+				if err != nil {
+					return
+				}
+				podLogString = strings.TrimSpace(buf.String())
+			}()
+		}
+	}
+	if podLogString != expectedPodlogString {
+		return podLogString, errors.New(err.Error())
```

Comment:
This will throw a nil pointer dereference if nothing in the inner function returns an error but the pod container logs from the pod are not equal to the pod container logs the test case expected.

I'd suggest creating a new error object with a new custom error message string.

I think you need a separate check before this check to return `podLogString, err` if `err` is not nil.